train_script.py

```diff
@@ -210,29 +215,47 @@
     avg_path = './data/predict-hd/linearaverageof100.npz'
     n_condns = 3
 
+# elif dataset == 'oasis':
+#     fpath = './data/oasis/train/*.npz'
+#     avg_path = './data/oasis/linearaverage.npz'
+#     n_condns = 1
+
 else:
-    raise ValueError('dataset expected to be dHCP or pHD')
+    fpath = args.fpath
+    avg_path = args.avg_path
+    n_condns = args.n_condns
+    if fpath is None or avg_path is None:
+        raise ValueError('Please provide fpath and avg_path for custom data.')
+    print("Using custom data with fpath: {}".format(fpath))
 
 
 img_paths = glob.glob(fpath)
 
+# use an element of the dataset to figure out the shape of the volume
+vol_shape = list(np.load(img_paths[0])['vol'].shape)
+
 Dtrain_data_generator = D_data_generator(
-    vol_shape=(160, 192, 160),
+    # vol_shape=(160, 192, 160),
+    vol_shape=vol_shape,
     img_list=img_paths,
     oversample_age=oversample,
     batch_size=batch_size,
     dataset=dataset,
 )
 
 Gtrain_data_generator = G_data_generator(
-    vol_shape=(160, 192, 160),
+    # vol_shape=(160, 192, 160),
+    vol_shape=vol_shape,
     img_list=img_paths,
     oversample_age=oversample,
     batch_size=batch_size,
     dataset=dataset,
 )
 
-avg_img = np.load(avg_path)['arr_0']  # TODO: make generic fname in npz
+try:
+    avg_img = np.load(avg_path)['arr_0']  # TODO: make generic fname in npz
+except:
+    avg_img = np.load(avg_path)['vol']
```

Comment:
should probably be:
```
try:
    avg_img = np.load(avg_path)['vol']
except:
    avg_img = np.load(avg_path)['arr_0']
```
as `arr_0` is the default name if left undefined in an npz iirc